test/Dialect/Substrait/literal.mlir

```diff
@@ -227,3 +227,18 @@ substrait.plan version 0 : 42 : 1 {
     yield %1 : tuple<si1, si1, si8, si16, si32, si64>
   }
 }
+
+// -----
+
+// CHECK:         %[[VAL_8:.*]] = substrait.literal #substrait.decimal<"1234567.89" : !substrait.decimal<9, 2>>
+// CHECK:         %[[VAL_9:.*]] = substrait.literal #substrait.decimal<"0.123" : !substrait.decimal<3, 3>>
+// CHECK:         %[[VAL_10:.*]] = substrait.literal #substrait.decimal<"0.123" : !substrait.decimal<3, 3>>
+// CHECK:         %[[VAL_11:.*]] = substrait.literal #substrait.decimal<"123.0" : !substrait.decimal<3, 0>>
+// CHECK:         %[[VAL_12:.*]] = substrait.literal #substrait.decimal<"123.0" : !substrait.decimal<3, 0>>
+// CHECK:         %[[VAL_13:.*]] = substrait.literal #substrait.decimal<"1111111111111111111111111111111111113.2" : !substrait.decimal<38, 1>>
+%d0 = substrait.literal #substrait.decimal<"1234567.89" : !substrait.decimal<9, 2>>
+%d1 = substrait.literal #substrait.decimal<"0.123" : !substrait.decimal<3, 3>>
+%d2 = substrait.literal #substrait.decimal<"000.123" : !substrait.decimal<3, 3>>
+%d3 = substrait.literal #substrait.decimal<"123.0" : !substrait.decimal<3, 0>>
+%d4 = substrait.literal #substrait.decimal<"123.000" : !substrait.decimal<3, 0>>
+%d5 = substrait.literal #substrait.decimal<"1111111111111111111111111111111111113.2" : !substrait.decimal<38, 1>>
```

Comment:
Nice! This is so much simpler than the whole plans that we have used until now!

Reply:
Yea, i was wondering why this file wasn't just using the literals directly instead of nesting them inside a relation... perhaps this file was initially written when `substrait.literal` had stricter parent op requirements?